cmake/targets/ara/araTarget.cmake

```diff
@@ -53,9 +53,8 @@ MACRO(ADD_EXECUTABLE_ARA_INTERNAL TARGET_NAME ADD_PLATFORM_FILES)
     ADD_EXECUTABLE(${TARGET_NAME} ${SRC_FILES})
 
     IF(NOT ARA_MACRO_ALREADY_EXECUTED)
-        # this link might be useful https://stackoverflow.com/questions/73871879/adding-a-step-for-building-linker-script-file-from-template-in-cmake
         # the following is transferred from https://github.com/pulp-platform/ara/blob/main/apps/Makefile#L48-L53
-        ADD_CUSTOM_COMMAND(TARGET ${TARGET_NAME}
+        ADD_CUSTOM_COMMAND(TARGET ${PROJECT_NAME} # Is the PROJECT_NAME here OK?
```

Comment:
Not really. Instead you want to a a new target consuming the generated linker file like this:
```
add_custom_target(my_linkerscript
  DEPENDS ${CMAKE_SOURCE_DIR}/my_linkerscript.ld) 
```

Finally you have to tell CMake that the linker script needs to be generated before the final executable is build/linked. This can be achieved as follows:

`add_dependencies(${PROJECT_NAME} my_linkerscript) `